runtime/src/bank.rs

```diff
@@ -403,19 +403,17 @@ impl Bank {
                         );
                     }
                 }
-                Err(TransactionError::BlockhashNotFound) => (),
-                Err(TransactionError::DuplicateSignature) => (),
-                Err(TransactionError::AccountNotFound) => (),
-                Err(e) => {
+                Err(TransactionError::InstructionError(b, e)) => {
                     if !tx.signatures.is_empty() {
                         status_cache.insert(
                             &tx.message().recent_blockhash,
                             &tx.signatures[0],
                             self.slot(),
-                            Err(e.clone()),
+                            Err(TransactionError::InstructionError(*b, e.clone())),
                         );
                     }
                 }
+                Err(_) => (),
```

Comment:
minor: maybe we should be explicit about these (use intellij to the rest if you want) ?

Reply:
If we add an error, I want the default behavior to be not to record the error so that we don't run into this again